crates/provider/src/kubernetes/node.rs

```diff
@@ -293,6 +345,40 @@ where
             .unwrap_or_else(|| panic!("namespace shouldn't be dropped, {}", THIS_IS_A_BUG))
     }
 
+    async fn upload_to_fileserver(&self, location: &Path) -> Result<Url, ProviderError> {
+        let data = self.filesystem.read(location).await?;
+        let hashed_path = hex::encode(sha2::Sha256::digest(&data));
+        let req = self
+            .http_client
+            .head(format!(
+                "http://{}/{hashed_path}",
+                self.file_server_local_host().await?
+            ))
+            .build()
+            .map_err(|err| {
+                ProviderError::UploadFile(location.to_string_lossy().to_string(), err.into())
+            })?;
+
+        let url = req.url().clone();
+        let res = self.http_client.execute(req).await.map_err(|err| {
+            ProviderError::UploadFile(location.to_string_lossy().to_string(), err.into())
+        })?;
+
+        if res.status() != reqwest::StatusCode::OK {
+            // we need to upload the file
+            self.http_client
+                .post(url.as_ref())
+                .body(data)
+                .send()
+                .await
+                .map_err(|err| {
```

Comment:
What is the difference with ```send_file``` below ? You want to update without copying to the pod ? Then maybe split the logic below in 2 parts and have a function used where we need ? wdyt ?

Reply:
Yes, sounds good. We should split the send_file (upload/download) since we don't always want to just copy to one pod.